src/USER-HMA/compute_hma.cpp

```diff
@@ -0,0 +1,473 @@
+
+ComputeHMA::ComputeHMA(LAMMPS *lmp, int narg, char **arg) :
+  Compute(lmp, narg, arg), id_temp(NULL), deltaR(NULL)
+{
+  if (narg < 4) error->all(FLERR,"Illegal compute hma command");                    
+  if (igroup) error->all(FLERR,"Compute hma must use group all");
+  if (strcmp(arg[3],"NULL") == 0) {error->all(FLERR,"fix ID specifying the set temperature of canonical simulation is required");}       
+  else {
+    int n = strlen(arg[3]) + 1;                  
+    id_temp = new char[n];   
+    strcpy(id_temp,arg[3]);                         
+  }
+ 
+  create_attribute = 1;                  
+  extscalar = 1;                         
+  timeflag = 1;                         
+
+  // (from compute displace/atom) create a new fix STORE style  
+  // our new fix's id (id_fix)= compute-ID + COMPUTE_STORE 
+  // our new fix's group = same as compute group
+
+  int n = strlen(id) + strlen("_COMPUTE_STORE") + 1;
+  id_fix = new char[n];
+  strcpy(id_fix,id);
+  strcat(id_fix,"_COMPUTE_STORE");
+
+  char **newarg = new char*[6];     
+  newarg[0] = id_fix;
+  newarg[1] = group->names[igroup];
+  newarg[2] = (char *) "STORE";              
+  newarg[3] = (char *) "peratom";
+  newarg[4] = (char *) "1";
+  newarg[5] = (char *) "3";
+  modify->add_fix(6,newarg);        
+  fix = (FixStore *) modify->fix[modify->nfix-1];    
+ 
+  delete [] newarg;                               
+
+  // calculate xu,yu,zu for fix store array
+  // skip if reset from restart file
+
+  if (fix->restart_reset) fix->restart_reset = 0;   
+  else {
+    double **xoriginal = fix->astore;            
+    double **x = atom->x;
+    imageint *image = atom->image;
+    int nlocal = atom->nlocal;
+
+    for (int i = 0; i < nlocal; i++)
+      domain->unmap(x[i],image[i],xoriginal[i]);          
+  }
+
+  vector_flag = 1;
+  extvector = -1;
+  comm_forward = 0; // 3 if 2nd derivative needed
+
+  computeU = computeP = computeCv = -1;
+  returnAnharmonic = 0;
+  std::vector<int> extvec;
+  for (int iarg=4; iarg<narg; iarg++) {
+    if (!strcasecmp(arg[iarg], "u")) {
+      computeU = extvec.size();
+      extvec.push_back(1);
+    }
+    else if (!strcasecmp(arg[iarg], "p")) {
+      if (iarg+2 > narg) error->all(FLERR,"Illegal fix hma command");
+      computeP = extvec.size();
+      deltaPcap = force->numeric(FLERR, arg[iarg+1]);
+      extvec.push_back(0);
+      iarg++;
+    }
+    else if (!strcasecmp(arg[iarg], "cv")) {
+      computeCv = extvec.size();
+      comm_forward = 3;
+      extvec.push_back(1);
+    }
+    else if (!strcasecmp(arg[iarg], "anharmonic")) {
+      // the first time we're called, we'll grab lattice pressure and energy
+      returnAnharmonic = -1;
+    }
+    else {
+      error->all(FLERR,"Illegal fix hma command");
+    }
+  }
+
+  if (extvec.size() == 0) {
+    error->all(FLERR,"Illegal fix hma command");
+  }
+  size_vector = extvec.size();
+  memory->create(vector, size_vector, "hma::vector");
```

Comment:
i would avoid symbol names like `vector` that could collide with other commonly used identifiers (even if we use namespaces to reduce the likelihood of clashes, it may confuse people reading the code.

Reply:
hmm, do you mean the first argument to `create`?  That's the member of `Compute`.  Or do you mean the name (3rd argument)?  I was trying to following other other compute/fix classes, which seem to use that convention.  I could name it "hma:results" or something.

Reply:
i meant the first argument. if this is done in other fixes computes, then it is ok. especially if you remove the use of `std::vector`. i am certain the name was used in the rest of the code by somebody not aware, that there is a `vector` type in STL. while technically, there is no conflict thanks to namespaces, it'll get confusing when people add `using namespace std;` I may suggest a change at a later point to change this for all fixes that employ class members called `vector` or others that overlap with commonly used types.